pkg/trace/agent/obfuscate.go

```diff
@@ -110,9 +110,7 @@ func (a *Agent) obfuscateSpan(span *pb.Span) {
 	}
 }
 
-func (a *Agent) obfuscateStatsGroup(b *pb.ClientGroupedStats) {
-	o := a.lazyInitObfuscator()
-
+func obfuscateStatsGroup(o *obfuscate.Obfuscator, b *pb.ClientGroupedStats) {
```

Comment:
if we have other dependencies not in Agent using this function; I could break out this change to a helper function instead of changing the signature